.github/workflows/terraform-tests.yml

```
@@ -0,0 +1,80 @@
+on:
+  pull_request:
+    branches:
+      - master
+
+name: Terraform Tests
+jobs:
+
+  prepare_list:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(python tests/terraform/get-tf-services.py --names)"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  test_service:
+    needs: prepare_list
+    strategy:
+      fail-fast: false
+      matrix:
+        service: ${{ fromJson(needs.prepare_list.outputs.matrix) }}
+        go-version: [1.18.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      DNS_ADDRESS: 0
+      AWS_DEFAULT_REGION: us-east-2
+      AWS_ALTERNATE_REGION: eu-west-1
+
+    steps:
+    - uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - uses: actions/checkout@v3
+    - uses: actions/checkout@v3
```

nit: Since the step is necessary (see comment above), I would suggest to add a newline (to stress that this is on purpose and an individual step):
```suggestion
    - uses: actions/checkout@v3
    
    - uses: actions/checkout@v3
```

---

To make it easier to understand I've added names for this steps.